docs/theme.md

```diff
@@ -0,0 +1,275 @@
+}))
+
+```
+
+# Hook
+
+- useTheme
+- useBase
+
+### useTheme
+
+The `useTheme` hook allows you to access the theme object in any component without having to pass it down as a prop. Here is an example of how to use the `useTheme` hook:
+
+```ts
+import React from 'react'
+import {useTheme} from 'rn-base-component'
+
+const MyComponent = () => {
+  const theme = useTheme()
+
+  return (
+    <View style={{backgroundColor: theme.colors.cardPrimaryBackground}}>
+      {/* Your component content goes here */}
+    </View>
+  )
+}
+```
+
+### uesBase
+
+```ts
+{
+    theme: {...},
+    colorMode: 'light' | 'dark',
+    toggleColorMode: () => {},
+    setColorMode: (colorMode) => {},
+}
+```
+
+- The `colorMode` property in the `useBase` hook is used to determine the current color mode of the app. It can be set to 'light', 'dark', null, or undefined.
+- The `toggleColorMode` function is used to toggle between 'light' and 'dark' color modes.
+- The `setColorMode` function is used to set the color mode to a specific value.
+
+```ts
+import React from 'react'
+import {Button} from 'react-native'
+import {uesBase} from 'rn-base-component'
+
+const MyComponent = () => {
+  const {colorMode, toggleColorMode, setColorMode} = uesBase()
+
+  return (
+    <View>
+      <Text>{colorMode}</Text>
+      <Button title="Toggle Color" onPress={toggleColorModee} />
+      <Button title="Set Color Light" onPress={() => setColorMode('light')} />
+      <Button title="Set Color Dark" onPress={() => setColorMode('dark')} />
+    </View>
+  )
+}
+```
+
+# Default theme
+
+### Colors
+
+```ts
+const colors = {
+  mainBackground: '#F0F2F3',
+  cardPrimaryBackground: '#0A906E',
+  textColor: '#0B0B0B',
+}
+```
+
+### Typography
+
+```ts
+const typography = {
+  fontWeights: {
+    hairline: 100,
+    thin: 200,
+    light: 300,
+    normal: 400,
+    medium: 500,
+    semibold: 600,
+    bold: 700,
+    extrabold: 800,
+    black: 900,
+    extraBlack: 950,
+  },
+  fonts: {
+    heading: undefined,
+    body: undefined,
+    mono: undefined,
+  },
+  fontSizes: {
+    '2xs': 10,
+    xs: 12,
+    sm: 14,
+    md: 16,
+    lg: 18,
+    xl: 20,
+    '2xl': 24,
+    '3xl': 30,
+    '4xl': 36,
+    '5xl': 48,
+    '6xl': 60,
+    '7xl': 72,
+    '8xl': 96,
+    '9xl': 128,
+  },
+}
+```
+
+### Border Width
+
+```ts
+const borderWidths = {
+  smallest: 1,
+  tiny: 2,
+  small: 4,
+  medium: 6,
+  large: 8,
+  huge: 10,
+  enormous: 12,
+  massive: 14,
+  gargantuan: 16,
+  colossal: 18,
+  largest: 20,
+}
+```
+
+### Sizes
+
+```ts
+const sizes = {
+  tiny: responsiveHeight(4),
+  small: responsiveHeight(8),
+  medium: responsiveHeight(16),
+  large: responsiveHeight(20),
+  huge: responsiveHeight(24),
+  enormous: responsiveHeight(32),
+  massive: responsiveHeight(48),
+  gargantuan: responsiveHeight(64),
+  colossal: responsiveHeight(96),
+  gigantic: responsiveHeight(128),
+  extraLarge: responsiveHeight(160),
+  jumbo: responsiveHeight(200),
+}
+```
+
+### Spacing
```

Comment:
What metric is this Spacing used for? And why is responsiveHeight?

Reply:
updated